mobify/deckard
# Refactoring and code cleanup

Converting deckard to use AMD wrapper and some other refactorings that I bolted on to this PR :)

Status: **Ready to merge**

Reviewers: @tedtate @johnboxall 
JIRA: https://mobify.atlassian.net/browse/CSOPS-766
## Changes
- converted to use AMD wrapper
- removed orientation-changing class. It's not guaranteed to be removed when the event is completed, therefore provides a bit of a buggy implementation
- added examples section
- refactored detect and orientation functions to separate concerns; added addClasses function to perform class adding
- simplified class toggling on orientation change to reduce reflows
## How to test-drive this PR
- checkout this branch
- `grunt test`
- integrate into a plugin, like lockup, and ensure it works
- `grunt serve` and use the examples page at http://localhost:3000/examples/index.html to verify deckard is correctly adding classes/properties. Use Chrome dev tools to toggle devices and refresh to verify updates are occurring correctly.
